src/components/signals/SignalCard.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+import { Card, Col, List, Row, Typography, Button } from "antd";
+import { IWhiteflagContext } from "../../helpers/Context";
+import { DecodedSignal } from "../../models/DecodedSignal";
+import { InfrastructureSubjectCode } from "../../models/WhiteflagSignal";
+import { SignalBodyText } from "../../models/SignalBodyText";
+import { CompassOutlined, EnvironmentOutlined } from "@ant-design/icons";
+import dayjs from "dayjs";
+import { useNavigate } from "react-router-dom";
+import { formatCoordinate } from "../../helpers/CoordinatesHelper";
+
+interface SignalProps {
+  ctx: IWhiteflagContext;
```

Comment:
Remove this as a prop you can use UseContext instead, also use React.FC component

interface SignalCardProps {
  signal: DecodedSignal;
}

const SignalCard: React.FC<SignalCardProps> = ({signal}) => {
  const navigate = useNavigate();
  const ctx = useContext(WhiteFlagContext);
  const { latitude, longitude } = ctx.extractCoordinates(signal);
  const bearing = ctx.calculateBearing(signal);
  const subjectCodeIndex = Object.values(InfrastructureSubjectCode).indexOf(
    signal.signal_body.subjectCode
  );

  const texts = signal?.signal_body?.text
    ? (JSON.parse(signal.signal_body.text) as SignalBodyText)
    : undefined;

Reply:
The signal prop is actually necessary